Assets/MeshLocalIdFix/MeshLocalIdFix.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using UnityEditor;
+using UnityEngine;
+
+public static class MeshLocalIdFix
+{
+    // What is this ?
+    // We changed the way LocalFileID are generated between Unity 2018.3 and 2019.1,
+    // Unfortunately a bug was found later where Meshes with duplicated names in 2018.3
+    // would be imported with the new ID system in 2019.1 and thus all references to them got lost.
+    // We fixed that in 2019.2.2f1 so all duplicated Mesh that exists before 2019.1 would keep their old ID.
+
+    // The new emerging problem now is:
+    // If you had a Mesh in 2018.3, that you migrated to 2019.1 and started using it in this version,
+    // references to its Mesh would use the new ID format for duplicated mesh names.
+    // Upgrading to 2019.2.2f1 and newest will break all those references.
+    // There is no good way to fix that internally without breaking everyone else project.
+
+    // This script is going through all ModelImporter in the project and figure out possible mismatching ids.
+    // Use it with care and be sure to save your project first because this is a destructive action that cannot be reverted.
+    // Also, it may takes a lot of time to run
+    // a good suggestion would be to run it on a separate computer or over night using the following command line:
+    // unity.exe -quit -batchmode -executeMethod MeshLocalIdFix.DoFixReferences
+    // see https://docs.unity3d.com/Manual/CommandLineArguments.html for more information.
+    [MenuItem("Assets/FBX File ID/Fix All references in prefabs and scenes")]
+    static void DoFixReferences()
+    {
+        // Ask for backup first
+        if (!EditorUtility.DisplayDialog("Fixing Mesh IDs from 2019.1 migration",
+            "This operation is going through all Scenes and Prefabs and cannot be reverted.\nHave you made a backup?",
+            "I'm fine",
+            "Cancel"))
+            return;
+
+        // AssetImporter.MakeLocalFileIDWithHash is internal, to avoid looking for it on every iteration (like, a lot)
```

Comment:
// AssetImporter.MakeLocalFileIDWithHash is internal, to avoid looking it up on every iteration (like, a lot)
        // We create a delegate once and use it later in the script.